src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1831,11 +1831,12 @@ private BoundExpression BindNameofOperatorInternal(InvocationExpressionSyntax no
         {
             CheckFeatureAvailability(node, MessageID.IDS_FeatureNameof, diagnostics);
             var argument = node.ArgumentList.Arguments[0].Expression;
-            string name = "";
             // We relax the instance-vs-static requirement for top-level member access expressions by creating a NameofBinder binder.
             var nameofBinder = new NameofBinder(argument, this);
             var boundArgument = nameofBinder.BindExpression(argument, diagnostics);
-            if (!boundArgument.HasAnyErrors && CheckSyntaxForNameofArgument(argument, out name, diagnostics) && boundArgument.Kind == BoundKind.MethodGroup)
+
+            bool syntaxIsOk = CheckSyntaxForNameofArgument(argument, out string name, boundArgument.HasAnyErrors ? BindingDiagnosticBag.Discarded : diagnostics);
```

Comment:
I didn't want to introduce a local for this conditional expression to prevent it from being used accidentally. (See few lines later which uses `diagnostics` regardless `syntaxIsOk`)